lib/rules/padding-line-between-statements.js

```diff
@@ -0,0 +1,585 @@
+/**
+ * @fileoverview Rule to require or disallow newlines between statements
+ * @author Toru Nagashima
+ */
+
+"use strict";
+
+//------------------------------------------------------------------------------
+// Requirements
+//------------------------------------------------------------------------------
+
+const astUtils = require("../ast-utils");
+
+//------------------------------------------------------------------------------
+// Helpers
+//------------------------------------------------------------------------------
+
+const VALID_PARENT = /^(?:Program|(?:Switch|Block)Statement|SwitchCase)$/;
```

Comment:
This has `SwitchStatement` also, so I need to separate the check of `SwitchStatement`.